src/model.h

```diff
@@ -110,6 +116,50 @@ struct RapidsModel : rapids::Model<RapidsSharedState> {
     }
 
     output.finalize();
+
+    /* For running gpu treeshap
+     * Make sure that input and output buffers are in device memory
+     * for gpu treeshap */
+    if constexpr (rapids::IS_GPU_BUILD && IS_TREESHAP_BUILD) {
+      // boolean to check whether gpu treeshap will be executed
+      auto run_gpu_treeshap =
+          shared_state->check_output_name("treeshap_output");
+
+      if (run_gpu_treeshap) {
+        auto treeshap_output = get_output<float>(batch, "treeshap_output");
+        auto treeshap_output_buffer = rapids::Buffer<float>(
+            treeshap_output.data(), treeshap_output.size(),
+            treeshap_output.mem_type(), treeshap_output.device(),
+            treeshap_output.stream());
+
+        if (gpu_treeshap_model.has_value()) {
+          // Always copy input buffer to device memory
+          if (input.mem_type() == rapids::HostMemory &&
+              input_buffer.mem_type() == rapids::HostMemory) {
+            input_buffer = rapids::Buffer<float const>(
```

Comment:
Could you add a `TODO` just saying that we should look at collapsing this `input_buffer` with the one used in ordinary inference? The logic is a little tricky, so I don't want us to do it right now, but doing so would save us a host-to-device copy.

Reply:
What do you mean by collapsing? GPUTreeshap runs only on device memory, so I'm not sure how we can avoid host-to-device copies